testsys/src/run_aws_ecs.rs

```diff
@@ -209,44 +402,69 @@ impl RunAwsEcs {
                             task_count: self.task_count,
                             subnet: format!("${{{}.publicSubnetId}}", cluster_resource_name),
                             task_definition_name_and_revision: self
-                                .task_definition_name_and_revision,
+                                .task_definition_name_and_revision
+                                .clone(),
                         }
                         .into_map()
                         .context(error::ConfigMapSnafu)?,
                     ),
-                    secrets: aws_secret_map,
+                    secrets,
                     capabilities: None,
                 },
             },
             status: None,
-        };
-        let resource_client = ResourceClient::new_from_k8s_client(k8s_client.clone());
-        let test_client = TestClient::new_from_k8s_client(k8s_client);
-
-        let _ = resource_client
-            .create(ecs_resource)
-            .await
-            .context(error::ModelClientSnafu {
-                message: "Unable to create ECS cluster resource object",
-            })?;
-        println!("Created resource object '{}'", cluster_resource_name);
-
-        let _ = resource_client
-            .create(ec2_resource)
-            .await
-            .context(error::ModelClientSnafu {
-                message: "Unable to create EC2 instances resource object",
-            })?;
-        println!("Created resource object '{}'", ec2_resource_name);
-
-        let _ = test_client
-            .create(test)
-            .await
-            .context(error::ModelClientSnafu {
-                message: "Unable to create test object",
-            })?;
-        println!("Created test object '{}'", self.name);
+        })
+    }
 
-        Ok(())
+    #[allow(clippy::too_many_arguments)]
+    fn migration_test(
+        &self,
+        name: &str,
+        version: &str,
+        tuf_repo: Option<TufRepoConfig>,
+        secrets: Option<BTreeMap<String, SecretName>>,
+        ec2_resource_name: &str,
+        cluster_resource_name: &str,
+        depends_on: Option<Vec<String>>,
+        migration_agent_image: &str,
+        migration_agent_pull_secret: &Option<String>,
```

Comment:
We should try removing the options around things that are actually not optional. Below in the function body you can say `Some(tuf_repo)`. The guiding principles are: a) validate data (i.e. deal with missing data) as early as possible, and b) making function signatures that cannot be misused.

Reply:
The tuf repo is actually optional.